prowler/lib/outputs/finding.py

```diff
@@ -267,3 +276,179 @@ def generate_output(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
             raise error
+
+    @classmethod
+    def transform_api_finding(cls, finding, provider) -> "Finding":
+        """
+        Transform a FindingModel instance into an API-friendly Finding object.
+
+        This class method extracts data from a FindingModel instance and maps its
+        properties to a new Finding object. The transformation populates various
+        fields including authentication details, timestamp, account information,
+        check metadata (such as provider, check ID, title, type, service, severity,
+        and remediation details), as well as resource-specific data. The resulting
+        Finding object is structured for use in API responses or further processing.
+
+        Args:
+            finding (API Finding): An API Finding instance containing data from the database.
+            provider (Provider): the provider object.
+
+        Returns:
+            Finding: A new Finding instance populated with data from the provided model.
+        """
+        # Missing Finding's API values
+        finding.muted = False
+        finding.resource_details = ""
+        resource = finding.resources.first()
+        finding.resource_arn = resource.uid
+        finding.resource_name = resource.name
+
+        finding.resource = model_to_dict(resource)
```

Comment:
```suggestion
        finding.resource = {}
```

I'd exclude this as it is only done for the OCSF in the SDK because the API has not all the values that this field is intended to have, I prefer to add a TODO and work on it on the future once we have the full resource data into the Prowler API DB.